ktlint/src/main/kotlin/com/pinterest/ktlint/internal/FileUtils.kt

```diff
@@ -40,6 +40,7 @@ internal val DEFAULT_PATTERNS = DEFAULT_KOTLIN_FILE_EXTENSIONS.map { "**/*.$it"
 internal fun FileSystem.fileSequence(
     patterns: List<String>,
     rootDir: Path = Paths.get(".").toAbsolutePath().normalize(),
+    enableDefaultPatterns: Boolean = true,
```

Comment:
I dislike boolean parameters. Also, I doubt whether an additional parameter is needed. The `patterns` can be set to `DEFAULT_PATTERNS` in 'KtlintCommandLine` except when `--patterns-from-stdin` is set.

Reply:
Adding the default patterns in `KtlintCommandLine` does sound better, but the problem is that `FileSystem.fileSequence()` adds the default patterns itself when no patterns are passed in.
If we don't wanna change the signature of `FileSystem.fileSequence()`, then we'd need to change it so that it doesn't add the default patterns at all:

```diff
diff --git a/ktlint/src/main/kotlin/com/pinterest/ktlint/internal/FileUtils.kt b/ktlint/src/main/kotlin/com/pinterest/ktlint/internal/FileUtils.kt
index 61b3a92c..51614f60 100644
--- a/ktlint/src/main/kotlin/com/pinterest/ktlint/internal/FileUtils.kt
+++ b/ktlint/src/main/kotlin/com/pinterest/ktlint/internal/FileUtils.kt
@@ -60,15 +60,9 @@ internal fun FileSystem.fileSequence(
 
     val globs = expand(patternsExclusiveExistingFiles, rootDir)
 
-    val pathMatchers = if (globs.isEmpty()) {
-        DEFAULT_PATTERNS
-            .map { getPathMatcher("glob:$it") }
-            .toSet()
-    } else {
-        globs
-            .filterNot { it.startsWith(NEGATION_PREFIX) }
-            .map { getPathMatcher(it) }
-    }
+    val pathMatchers = globs
+        .filterNot { it.startsWith(NEGATION_PREFIX) }
+        .map { getPathMatcher(it) }
 
     val negatedPathMatchers = if (globs.isEmpty()) {
         emptySet()

```

But this change breaks [a test in `FileUtilsTest.kt`](https://github.com/pinterest/ktlint/blob/d51de4cb14473ed5cbb732039585d5247c7af2de/ktlint/src/test/kotlin/com/pinterest/ktlint/internal/FileUtilsTest.kt#L68-L85).
Do we just remove that test?

Reply:
The signature of the fileSequence function can be changed as it is an internal function. So it does not break any API.